AutoRest/Generators/Ruby/Azure.Ruby.Tests/RspecTests/lro_spec.rb

```diff
@@ -8,221 +8,433 @@
+    result = @lros_client.put200succeeded(@product).value!
     expect(result.body.provisioning_state).to eq("Succeeded")
   end
 
   it 'should serve success responce on initial request without provision state' do
-    result = @client.lros.put200succeeded_no_state(@product).value!
+    result = @lros_client.put200succeeded_no_state(@product).value!
     expect(result.body.id).to eq("100")
     expect(result.body.provisioning_state).to eq(nil)
   end
 
   it 'should serve 202 on initial responce and status responce without provision state' do
-    result = @client.lros.put202retry200(@product).value!
+    result = @lros_client.put202retry200(@product).value!
     expect(result.body.id).to eq("100")
     expect(result.body.provisioning_state).to eq(nil)
   end
 
-  it 'should retry on 500 server responce in PUT request' do
-    result = @client.lroretrys.put_async_relative_retry_succeeded(@product).value!
+  it 'should serve success responce on initial DELETE request' do
+    result = @lros_client.delete204succeeded().value!
+    expect(result.response.status).to eq(204)
+  end
+
+  it 'should return payload on POST async request' do
+    result = @lros_client.post200with_payload().value!
+    expect(result.body.id).to eq(1)
+  end
+
+  it 'should succeed for put async retry' do
+    result = @lros_client.put_async_retry_succeeded(@product).value!
+    expect(result.response.status).to eq(200)
     expect(result.body.provisioning_state).to eq("Succeeded")
   end
 
-  # TODO: Fix flakey test
-  #it 'should serve async PUT operation failed' do
-  #  expect { @client.lrosads.put_async_relative_retry400_async(@product).value! }.to raise_exception(MsRestAzure::AzureOperationError)
-  #end
+  it 'should succeed for put async no retry' do
+    result = @lros_client.put_async_no_retry_succeeded(@product).value!
+    expect(result.response.status).to eq(200)
+  end
 
-  it 'should serve success responce on initial DELETE request' do
-    result = @client.lros.delete204succeeded().value!
+  it 'should fail for put async retry' do
+    expect { @lros_client.put_async_retry_failed(@product).value! }.to raise_error(MsRestAzure::AzureOperationError)
+  end
+
+  it 'should fail for put async no retry canceled' do
+    expect { @lros_client.put_async_no_retrycanceled(@product).value! }.to raise_error(MsRestAzure::AzureOperationError)
+  end
+
+  it 'should succeed for post async retry' do
+    result = @lros_client.post_async_retry_succeeded(@product).value!
+    expect(result.response.status).to eq(200)
+  end
+
+  it 'should succeed for post async no retry' do
+    result = @lros_client.post_async_no_retry_succeeded(@product).value!
+    expect(result.response.status).to eq(200)
+  end
+
+  it 'should fail for post async retry' do
+    expect { @lros_client.post_async_retry_failed(@product).value! }.to raise_error(MsRestAzure::AzureOperationError)
+  end
+
+  it 'should fail for post async no retry canceled' do

```

Comment:
yup corrected

---
In reply to: [69022094](https://github.com/Azure/autorest/pull/1188#discussion_r69022094) [](ancestors = 69022094)